sdk/src/test/java/com/qonversion/android/sdk/internal/userProperties/UserPropertiesStorageImplTest.kt

```diff
@@ -0,0 +1,451 @@
+            val keys = listOf("keyToDelete1", "keyToDelete2")
+
+            val existingProperties =
+                mutableMapOf("keyToDelete1" to "value1", "keyToDelete2" to "value2")
+
+            every {
+                spykStorage.putPropertiesToStorage()
+            } just runs
+
+            every {
+                spykStorage.properties
+            } returns existingProperties
+
+            // when
+            spykStorage.delete(keys)
+
+            // then
+            verify(exactly = 1) {
+                spykStorage.putPropertiesToStorage()
+            }
+            val updatedProperties = spykStorage.properties
+            assertThat(updatedProperties).isEmpty()
+        }
+    }
+
+    @Nested
+    inner class GetPropertiesFromStorage {
+        @Test
+        fun `get properties from filled storage`() {
+            // given
+            val jsonString = "jsonString"
+            val properties = mapOf("key1" to "value1", "key2" to "value2")
+            every {
+                mockLocalStorage.getString(propertiesKeyInMemory)
+            } returns jsonString
+
+            every {
+                mockMapper.toMap(jsonString)
+            } returns properties
+
+            // when
+            val result = userPropertiesStorage.getPropertiesFromStorage()
+
+            //then
+            assertThat(result).isEqualTo(result)
+        }
+
+        @Test
+        fun `get properties from empty storage`() {
+            // given
+            every {
+                mockLocalStorage.getString(propertiesKeyInMemory)
+            } returns null
+
+            // when
+            val result = userPropertiesStorage.getPropertiesFromStorage()
+
+            //then
+            verify { mockMapper wasNot called }
+            assertThat(result).isEmpty()
+        }
+
+        @Test
+        fun `get properties when exception occurred`() {
+            // given
+            val errorString = "Couldn't load properties from storage"
+            val jsonString = "jsonString"
+            every {
+                mockLocalStorage.getString(propertiesKeyInMemory)
+            } returns jsonString
+
+            val exception = IllegalStateException("Couldn't create JSONObject from string")
+            every {
+                mockMapper.toMap(jsonString)
+            } throws exception
+
+            // when
+            userPropertiesStorage.getPropertiesFromStorage()
+            assertThat(slotErrorLogMessage.captured)
+                .startsWith(errorString)
+            verify(exactly = 1) {
+                mockLogger.error(errorString, exception)
+            }
+        }
+    }
+
+    @Nested
+    inner class PutPropertiesToStorage {
+        @Test
+        fun `put properties from filled storage`() {
+            // given
+            val jsonString = "jsonString"
+            val existingProperties = mapOf("key1" to "value1", "key2" to "value2")
+
+            mockGetPropertiesFromStorage(existingProperties)
```

Comment:
Why don't you just spyk `properties` for these tests? I think it can be used for all tests but the getter ones